homeassistant/components/cast/config_flow.py

```diff
@@ -102,35 +130,58 @@ async def async_step_options(self, user_input=None):
         errors = {}
         current_config = self.config_entry.data
         if user_input is not None:
-            bad_hosts = False
+            bad_cec, ignore_cec = _string_to_list(
+                user_input.get(CONF_IGNORE_CEC, ""), IGNORE_CEC_SCHEMA
+            )
+            bad_hosts, known_hosts = _string_to_list(
+                user_input.get(CONF_KNOWN_HOSTS, ""), KNOWN_HOSTS_SCHEMA
+            )
+            bad_uuid, wanted_uuid = _string_to_list(
+                user_input.get(CONF_UUID, ""), WANTED_UUID_SCHEMA
+            )
 
-            known_hosts = user_input.get(CONF_KNOWN_HOSTS, "")
-            known_hosts = [x.strip() for x in known_hosts.split(",") if x.strip()]
-            try:
-                known_hosts = KNOWN_HOSTS_SCHEMA(known_hosts)
-            except vol.Invalid:
-                errors["base"] = "invalid_known_hosts"
-                bad_hosts = True
-            if not bad_hosts:
+            if not bad_cec and not bad_hosts and not bad_uuid:
                 updated_config = {}
+                updated_config[CONF_IGNORE_CEC] = ignore_cec
                 updated_config[CONF_KNOWN_HOSTS] = known_hosts
+                updated_config[CONF_UUID] = wanted_uuid
                 self.hass.config_entries.async_update_entry(
                     self.config_entry, data=updated_config
                 )
                 return self.async_create_entry(title="", data=None)
 
         fields = {}
-        known_hosts_string = ""
-        if current_config.get(CONF_KNOWN_HOSTS):
-            known_hosts_string = ",".join(current_config.get(CONF_KNOWN_HOSTS))
-        fields[
-            vol.Optional(
-                "known_hosts", description={"suggested_value": known_hosts_string}
-            )
-        ] = str
+        suggested_value = _list_to_string(current_config.get(CONF_KNOWN_HOSTS))
+        _add_with_suggestion(fields, CONF_KNOWN_HOSTS, suggested_value)
+        suggested_value = _list_to_string(current_config.get(CONF_UUID))
+        _add_with_suggestion(fields, CONF_UUID, suggested_value)
+        suggested_value = _list_to_string(current_config.get(CONF_IGNORE_CEC))
+        _add_with_suggestion(fields, CONF_IGNORE_CEC, suggested_value)
```

Comment:
We know if this flow is requested by an advanced user. Let's hide UUID and IGNORE CEC unless users are advanced.

Reply:
Fixed in 4487596